File: Assignment_1/app/src/main/java/com/kaushiknsanji/learndagger/ui/MainViewModel.java
package com.kaushiknsanji.learndagger.ui;

import com.kaushiknsanji.learndagger.data.local.DatabaseService;
import com.kaushiknsanji.learndagger.data.remote.NetworkService;
import com.kaushiknsanji.learndagger.di.scope.ActivityScope;

import javax.inject.Inject;

/**
 * Dummy ViewModel class for the {@link MainActivity}
 */
@ActivityScope
public class MainViewModel {

    private DatabaseService databaseService;
    private NetworkService networkService;

    @Inject
    public MainViewModel(DatabaseService databaseService, NetworkService networkService) {
        this.databaseService = databaseService;
        this.networkService = networkService;
    }

    public String getSomeData() {
        return databaseService.getDummyData() + " : " + networkService.getDummyData();
    }
}

File: Assignment_2/app/src/main/java/com/kaushiknsanji/learndagger/ui/main/MainViewModel.kt
package com.kaushiknsanji.learndagger.ui.main

import com.kaushiknsanji.learndagger.data.local.DatabaseService
import com.kaushiknsanji.learndagger.data.remote.NetworkService
import com.kaushiknsanji.learndagger.di.ActivityScope
import javax.inject.Inject

/**
 * Dummy ViewModel class for the [MainActivity]
 *
 * @author Kaushik N Sanji
 */
@ActivityScope
class MainViewModel @Inject constructor(
    private val databaseService: DatabaseService,
    private val networkService: NetworkService
) {

    /**
     * Method that returns some dummy data
     *
     * @return String containing some dummy data
     */
    fun getSomeData(): String = "${databaseService.getDummyData()} : ${networkService.getDummyData()}"

}